app/javascript/components/map/legend/legend-years/component.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './styles.scss';
+
+class PopupComponent extends React.Component {
+  static propTypes = {
+    activeLayer: PropTypes.object.isRequired,
+    onChangeYear: PropTypes.func.isRequired
+  }
+
+  onChange = (e) => {
+    const { activeLayer } = this.props;
+    this.props.onChangeYear(activeLayer.id, e.currentTarget.value);
+  }
+
+  render() {
+    const { activeLayer } = this.props;
+    const { year, years = [] } = activeLayer;
+
+    return (
+      <div className="c-legend-years">
+        {years.map((y) => {
+          const checked = year === y;
+
+          return (
+            <div key={y} className="legend-years--radio">
+              <input id={`radio-${activeLayer.type_id}-${y}`} type="radio" name={`years-${activeLayer.type_id}`} value={y} checked={checked} onChange={this.onChange} />
+              <label htmlFor={`radio-${activeLayer.type_id}-${y}`}>
```

Comment:
Is not the label previous to the input?

Reply:
no, because if you check the html you will see that we are looking for `input + label`. That means that it should be a brother and it MUST be below it